platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/tipset/TipsetEventCreationRulesTests.java

```diff
@@ -291,4 +296,41 @@ void rateLimitTest() {
             }
         }
     }
+
+    @Test
+    @DisplayName("ReconnectSavedStateRule Test")
+    void reconnectSavedStateRuleTest() {
+        final AtomicLong lastReconnectRound = new AtomicLong(NO_ROUND);
+        final AtomicLong lastSavedRound = new AtomicLong(NO_ROUND);
+
+        final ReconnectStateSavedRule rule = new ReconnectStateSavedRule(lastReconnectRound::get, lastSavedRound::get);
+
+        // No reconnects or state saves done yet
+        assertTrue(rule.isEventCreationPermitted());
+
+        // State saved, no reconnects
+        lastSavedRound.set(NO_ROUND);
```

Comment:
Presumably you meant to set `lastReconnectRound` to `NO_ROUND`, rather than setting `lastSavedRound` twice